### PR TITLE
[WIP] Add integration test showing possible ResourceQuota race

### PR DESCRIPTION
**Pull request by nojnhuh**
#### What type of PR is this?
/kind bug
/kind flake

#### What this PR does / why we need it:

This PR adds a (failing) integration test that reliably triggers the bug causing the e2e flake described by #128410. I intend to iterate on a fix for the issue in this PR as I continue to dig into why this is failing.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/quota/quota_test.go [hunk 1]
```diff
@@ -593,3 +597,119 @@ func newService(name string, svcType v1.ServiceType, allocateNodePort bool) *v1.
...
+	close(informersStarted)
+
+	wg := new(sync.WaitGroup)
+	namespaces := 500
+	for range namespaces {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			ns := framework.CreateNamespaceOrDie(clientset, "quota-"+utilrand.String(5), t)
```
Comment (nojnhuh) [inline test/integration/quota/quota_test.go:654]: @pohly My intent with this test was to mimic the flaky e2e test, but essentially running a bunch of those concurrently in different namespaces. Does this seem like a fair representation of that?

**Comment by nojnhuh** [comment]
Looking into the failures more, I noticed all of the ones that hit the same error condition as the e2e flake (where creating the second ResourceClaim succeeds) had hit some other error earlier in the test. I just pushed a change to iron out the control flow to fail early when that happens, and I no longer see any tests hit that same failure condition.

File: test/integration/quota/quota_test.go [hunk 2]
```diff
@@ -593,3 +596,204 @@ func newService(name string, svcType v1.ServiceType, allocateNodePort bool) *v1.
...
+	})
+	defer tearDownFn()
+
+	clientset := clientset.NewForConfigOrDie(kubeConfig)
+
+	informers := informers.NewSharedInformerFactory(clientset, controller.NoResyncPeriodFunc())
+
+	discoveryFunc := clientset.Discovery().ServerPreferredNamespacedResources
+	listerFuncForResource := generic.ListerFuncForResourceFunc(informers.ForResource)
+	qc := quotainstall.NewQuotaConfigurationForControllers(listerFuncForResource)
```
[the remark this answers is not in the document]
Comment (nojnhuh) [reply test/integration/quota/quota_test.go:620]: AFAICT that's all accurate. I don't see any extra synchronization that would prevent that kind of race.

I think I found a way to trigger this in a single namespace like the e2e test by adding the ResourceQuota to the API server's cache well after the controller manager's cache and that triggers the same failure as the flakes. Does this iteration of the test seem convincing?

[the remark this answers is not in the document]
Comment (pohly) [reply test/integration/quota/quota_test.go:620]: It all seems pretty clear now. We could try to get confirmation from SIG Auth that the quota mechanism isn't reliable.

The fix for the E2E flake then is to turn the one-shot creation of a second ResourceClaim into an "eventually we should get an error" when creating many ResourceClaims.

[the remark this answers is not in the document]
Comment (nojnhuh) [reply test/integration/quota/quota_test.go:620]: Thanks for following up on Slack! I've opened a PR with the `Eventually` workaround: #128943

**Comment by nojnhuh** [comment]
I'll close this for now, but we can reference it later if needed.

/close

**Pull request closed without merging**